### PR TITLE
Fix memory leak

### PR DESCRIPTION
**Pull request by sm6srw**
### Purpose

The logger constructors where called twice:
* This fixes the memory issue that caused the serial tests to timeout.

### Declarations

Check these if you believe they are true

- [X] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [X] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [X] This PR contains no files larger than 50 MB

### CODE REVIEW

File: src/DynamoCore/Logging/DynamoLogger.cs [hunk 1]
```diff
@@ -227,7 +227,6 @@ public DynamoLogger(DebugSettings debugSettings, string logDirectory, Boolean is
         /// <param name="isServiceMode">We want restrict logging in service mode to console only due to lambda limitations.</param>
         /// TODO(DYN-5757): Review usage of isTestMode,isTestMode,isServiceMode across Dynamo and see how we can consildate all these flags.
         public DynamoLogger(DebugSettings debugSettings, string logDirectory, Boolean isTestMode, Boolean isCLIMode, Boolean isServiceMode)
-            : this(debugSettings, logDirectory, isTestMode)
```
[the remark this answers is not in the document]
Comment (sm6srw) [reply src/DynamoCore/Logging/DynamoLogger.cs:230]: Yes

**Comment by sm6srw** [comment]
Parallel tests passed here: https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/8812/

**Comment by sm6srw** [comment]
@mjkkirschner PTAL

**Comment by mjkkirschner** [comment]
@sm6srw is that event unsubscribed anywhere?

**Comment by sm6srw** [comment]
Yes, it is. That was the leak. We subscribed twice but was only unsubscribing once.

**Pull request merged**